app/services/scraper/crawl_google_search_service.rb

```diff
@@ -3,48 +3,40 @@ class CrawlGoogleSearchService < ApplicationService
     GOOGLE_SEARCH_URL = 'https://www.google.com.vn/search?gl=us&q='.freeze
     IMPORT_CRAWL_RESULT_COLUMNS = %i[user_id keyword total_results search_time total_links total_ads source].freeze
 
-    def initialize(keywords, user_id, batch_id)
-      @keywords = keywords
+    def initialize(keyword, user_id, proxy_index)
+      @keyword = keyword
       @user_id = user_id
-      @batch_id = batch_id
-      @crawl_results = []
-      @failed_keywords = []
+      @proxy_index = proxy_index
     end
 
     def call
-      total_proxies = Settings.proxy.proxies.count
-      keywords.each.with_index do |keyword, index|
-        browser = Watir::Browser.new(:chrome, headless: true, proxy: proxy(index % total_proxies))
-        browser.goto("#{GOOGLE_SEARCH_URL}#{keyword}")
-        unless browser.element(id: 'result-stats').present? # rubocop:disable Rails/Blank
-          failed_keywords << keyword
-          next
-        end
-        crawl_results << crawl_data(keyword, browser.html)
-        browser.quit
-      rescue Net::ReadTimeout
-        failed_keywords << keyword
-      end
-
-      CrawlResult.import(IMPORT_CRAWL_RESULT_COLUMNS, crawl_results)
-      push_failed_keywords_to_cache
+      @crawl_result = CrawlResult.create(user_id: user_id, keyword: keyword, status: :pending)
+      crawl_data
+      create_notification if crawl_result.failed?
     end
 
   private
 
-    attr_reader :keywords, :user_id, :batch_id, :crawl_results, :failed_keywords
+    attr_reader :keyword, :user_id, :proxy_index, :crawl_result
 
-    def crawl_data(keyword, html)
-      doc = Nokogiri::HTML.parse(html)
-      {
-        user_id:,
-        keyword:,
+    def crawl_data
+      browser = Watir::Browser.new(:chrome, headless: true, proxy: proxy(proxy_index))
+      browser.goto("#{GOOGLE_SEARCH_URL}#{keyword}")
+
+      return crawl_result.update!(status: :failed) unless browser.element(id: 'result-stats').present? # rubocop:disable Rails/Blank
+
+      doc = Nokogiri::HTML.parse(browser.html)
+      crawl_result.update!(
```

Comment:
I recommend breaking the update in two operations to make the code easier to grap:

```
crawl_result.update_crawl_info(crawled_data)
crawl_result.mark_as_finished
```

It would be best to treat `CrawResult` as the entity and declare behaviors on it.